core/foundation/src/StringUtils.cxx

```diff
@@ -37,4 +38,25 @@ std::vector<std::string> Split(std::string_view str, std::string_view delims, bo
    return out;
 }
 
+/// Read and split into white-space separated components a (txt) file
+/// A component string starting with '#' will be ignored
+/// \param[in] str File to read
+/// \param[out] std::vector<std::string> A vector of strings
+std::vector<std::string> ReadFilelist (std::string filelist) {
+    std::vector<std::string> input_file_list;
+    std::fstream infile;
+    infile.open(filelist, std::ios::in);
+    
+    if (infile.is_open()) {
```

Comment:
If the file fails to open we should probably print an error message (something in the spirit of `::Error("func", "cannot file file %s", filename)`)?

Reply:
well, the intent of this function is to apply TChain::AddFile to a list of strings (for_each concept). if some error management is desired than it should be added to AddFile and as such out of scope of this PR.

Reply:
i added the Error message, and i just return the empty list